### PR TITLE
fix: eliminate `fakeDurable` and change global durability rules to match

### PR DESCRIPTION
**Pull request by FUDCo**
This addresses the SwingSet portion of #5593.

- Virtual objects and collections can no longer be declared `fakeDurable`; they are either durable or not, as was the case prior to the introduction of the `fakeDurable` feature.

- The `enableFakeDurable` config flag is renamed to `relaxDurabilityRules`.

- This renamed flag defaults to `false` when running in a swingset (as before) and to `true` when running on top of the mock virtuals library provided by `fakeVirtualSupport.js` (in the latter case it previously was unconditionally `true`; now it is settable as an option to `makeFakeVirtualStuff` et al).

- The meaning of the flag is altered.  By setting it to `true`, rather than allowing `fakeDurable` to be used or not, it instead globally alters the rules of what values can be admitted to durable objects and collections to be the same as their non-durable counterparts.

The expectation is that production code and tests that are exercising vat upgrade scenarios will run with the flag set to `false`, whereas tests that are exercising business logic that do not care about durability and which use mocks that won't pass the durability checks anyway will run with the flag set to `true`.  This expectation is reflected in the default flag values adopted by the kernel (`false`) and the fake virtual support code (`true`) respectively.

### CODE REVIEW

File: packages/SwingSet/src/liveslots/liveslots.js [hunk 1]
```diff
@@ -1446,6 +1445,11 @@ function build(
    * @returns {Promise<void>}
    */
   async function stopVat() {
+    assert(
+      !relaxDurabilityRules,
+      'stopVat not available when relaxDurabilityRules is true',
+    );
```
Comment (erights) [inline packages/SwingSet/src/liveslots/liveslots.js:1451]: Good!

(No change suggested)

**Review by erights — APPROVED** [review]
LGTM

I love seeing code melt away!

This feels right. But we'll see very soon ;)

File: packages/SwingSet/tools/fakeVirtualSupport.js [hunk 2]
```diff
@@ -246,14 +246,18 @@ export function makeFakeLiveSlotsStuff(options = {}) {
   };
 }
 
-export function makeFakeVirtualReferenceManager(fakeStuff) {
+export function makeFakeVirtualReferenceManager(
+  fakeStuff,
+  relaxDurabilityRules = true,
```
Comment (warner) [inline packages/SwingSet/tools/fakeVirtualSupport.js:251]: good, nice to have the option to tighten the rules again, even in a fake environment

**Review by warner — APPROVED** [review]
Looks good! At least from the swingset side, I'll defer to the others about whether it meets their needs.

**Pull request merged**